MockKit/MockPumpManager.swift

```diff
@@ -213,7 +220,16 @@ public final class MockPumpManager: TestingPumpManager {
             if oldStatus != newStatus {
                 notifyStatusObservers(oldStatus: oldStatus)
             }
-
+            
+            // stop insulin delivery as pump state requires
+            if (newValue.occlusionDetected != oldValue.occlusionDetected && newValue.occlusionDetected) ||
+                (newValue.pumpErrorDetected != oldValue.pumpErrorDetected && newValue.pumpErrorDetected) ||
+                (newValue.pumpBatteryChargeRemaining != oldValue.pumpBatteryChargeRemaining && newValue.pumpBatteryChargeRemaining == 0) ||
+                (newValue.reservoirUnitsRemaining != oldValue.reservoirUnitsRemaining && newValue.reservoirUnitsRemaining <= 0)
```

Comment:
Should insulin delivery be stopped if `case .suspended = state.suspendState`?

Reply:
It already gets stopped when the state is set to suspended.